### PR TITLE
executor: support the `SELECT INTO ... OUTFILE` syntax.

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve? 
Fix https://github.com/pingcap/tidb/issues/11112.
Support the `SELECT INTO ... OUTFILE` syntax.

### What is changed and how it works?
1. Introduce a new common plan `SelectInto` to let our planner support this syntax;
2. Introduce a new executor `SelectIntoExec`;
3. Add some tests.

### Check List 

Tests 

 - Unit test

### CODE REVIEW

**Comment by tiancaiamao** [comment]
Rest LGTM

**Comment by qw4990** [comment]
Thanks for your review @tiancaiamao , all comments have been addressed.

**Comment by qw4990** [comment]
PTAL @tiancaiamao @zz-jason

**Comment by tiancaiamao** [comment]
LGTM

File: executor/select_into.go [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+}
+
+// Open implements the Executor Open interface.
+func (s *SelectIntoExec) Open(ctx context.Context) error {
+	// only 'select ... into outfile' is supported now
+	if s.intoOpt.Tp != ast.SelectIntoOutfile {
+		return errors.New("unsupported SelectInto type")
+	}
+
+	f, err := os.Create(s.intoOpt.FileName)
```
Comment (SunRunAway) [inline executor/select_into.go]: From: https://dev.mysql.com/doc/refman/8.0/en/select-into.html
> file_name cannot be an existing file, which among other things prevents files such as /etc/passwd and database tables from being modified.

File: executor/select_into.go [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+		}
+		s.lineBuf = append(s.lineBuf, lineTerm...)
+		if _, err := s.writer.Write(s.lineBuf); err != nil {
+			return errors.Trace(err)
+		}
+	}
+	return nil
+}
+
+func (s *SelectIntoExec) dumpReal(row chunk.Row, col *expression.Column, colIdx int) {
```
Comment (SunRunAway) [inline executor/select_into.go]: Could you add a test covering this?

File: executor/select_into_test.go [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+}
+
+func (s *testSuite1) TestSelectIntoOutfileFromTable(c *C) {
+	tmpDir := os.TempDir()
+	outfile := path.Join(tmpDir, "select-into-outfile.data")
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("use test")
+
+	tk.MustExec("drop table if exists t")
+	tk.MustExec("create table t (i int, r real, d decimal(10, 5), s varchar(100), dt datetime, ts timestamp, j json)")
```
Comment (SunRunAway) [inline executor/select_into_test.go]: lack of `Duration`

**Comment by qw4990** [comment]
@SunRunAway Thanks for your review, all comments have been addressed now~ PTAL

File: executor/select_into.go (hunk 1 above)
Comment (SunRunAway) [reply executor/select_into.go]: Consider
```
os.OpenFile(name, O_RDWR | os.O_CREATE| os.O_EXCL, 0666)
```

File: executor/select_into_test.go [hunk 4]
```diff
@@ -15,11 +15,16 @@ package executor_test
 
 import (
 	"fmt"
+	"github.com/pingcap/tidb/executor"
```
Comment (SunRunAway) [inline executor/select_into_test.go]: Please group third party imports together.

File: executor/select_into.go (hunk 1 above)
Comment (qw4990) [reply executor/select_into.go]: Updated.

File: executor/select_into_test.go (hunk 4 above)
Comment (qw4990) [reply executor/select_into_test.go]: fixed, PTAL

**Comment by SunRunAway** [comment]
Please add the release note section in the PR description.  A PR of documentation is appreciated.

**Pull request merged**